src/java_bytecode/java_static_initializers.cpp

```diff
@@ -39,13 +67,35 @@ bool is_clinit_wrapper_function(const irep_idt &function_id)
   return has_suffix(id2string(function_id), clinit_wrapper_suffix);
 }
 
-/// Get name of the static-initialization-already-done global variable for a
-/// given class.
-/// \param class_name: class symbol name
-/// \return static initializer wrapper-already run global name
-static irep_idt clinit_already_run_variable_name(const irep_idt &class_name)
+/// Add a new symbol to the symbol table.
+/// Note: assumes that a symbol with this name does not exist.
+/// /param name: name of the symbol to be generated
+/// /param type: type of the symbol to be generated
+/// /param value: initial value of the symbol to be generated
+/// /param is_thread_local: if true this symbol will be set as thread local
+/// /param is_static_lifetime: if true this symbol will be set as static
+/// /return returns new symbol.
+static symbolt add_new_symbol(
```

Comment:
I (partially) agree to the comment above, I think this method should call `get_fresh_aux_symbol` to insert the new symbol in the symbol table, but that method does not initialize the remaining fields as this method does (`is_lvalue`, `value` etc) in the manner required in this file, so I would leave this method here?